src/BIP39Wallet.cs

```diff
@@ -0,0 +1,121 @@
+
+            public BlockchainWallet(string address, string privateKey, string mnemonic, string publicKey)
+            {
+                Address = address;
+                PrivateKey = privateKey;
+                Mnemonic = mnemonic;
+                PublicKey = publicKey;
+            }
+        }
+        public virtual string ConvertMnemonicToSeedHex(Mnemonic mnemonic, string password)
+        {
+            var mnemonicBytes = Encoding.UTF8.GetBytes(mnemonic.ToString().Normalize(NormalizationForm.FormKD));
+            var saltSuffix = string.Empty;
+            if (!string.IsNullOrEmpty(password))
+            {
+                saltSuffix = password;
+            }
+            var salt = $"mnemonic{saltSuffix}";
+            var saltBytes = Encoding.UTF8.GetBytes(salt);
+
+            var rfc2898DerivedBytes = new Rfc2898DeriveBytes(mnemonicBytes, saltBytes, 2048, HashAlgorithmName.SHA512);
+            var key = rfc2898DerivedBytes.GetBytes(64);
+            var hex = BitConverter
+                .ToString(key)
+                .Replace("-", "")
+                .ToLower();
+
+            return hex;
+        }
+
+        public virtual BlockchainWallet CreateWallet(int strength, Language language, string password)
+        {
+            var mnemonic = new Mnemonic(Wordlist.English, WordCount.Twelve);
+
+            var seed = ConvertMnemonicToSeedHex(mnemonic, "");
+            var masterKeyPath = new KeyPath("m/44'/1616'");
+            var masterWallet = new ExtKey(seed).Derive(masterKeyPath);
+            var wallet = masterWallet.Derive(new KeyPath("0'/0")).Derive(0);
+            var privateKey = wallet.PrivateKey;
+            var newKey = new Key(privateKey.ToBytes(), -1, false);
+            var publicKey = newKey.PubKey;
+
+            // Act
+            var address = Address.FromPublicKey(publicKey.ToBytes()).ToString().Trim('\"');
+            return new BlockchainWallet(address, privateKey.ToHex(), mnemonic.ToString(), publicKey.ToString());
+        }
+
+        public virtual BlockchainWallet GetWalletByMnemonic(string mnemonic, string password = "")
+        {
+            var mnemonicValue = new Mnemonic(mnemonic, Wordlist.English);
+            var seedHex = ConvertMnemonicToSeedHex(mnemonicValue, password);
+            var masterKeyPath = new KeyPath("m/44'/1616'");
+            var masterWallet = new ExtKey(seedHex).Derive(masterKeyPath);
+            var wallet = masterWallet.Derive(new KeyPath("0'/0")).Derive(0);
+            var privateKey = wallet.PrivateKey;
+            var newKey = new Key(privateKey.ToBytes(), -1, false);
+            var publicKey = newKey.PubKey;
+
+            // Act
+            var address = Address.FromPublicKey(publicKey.ToBytes()).ToString().Trim('\"');
+            return new BlockchainWallet(address, privateKey.ToHex(), mnemonic, publicKey.ToString());
+        }
+        
+        static byte[] StringToByteArray(string hexString)
```

Comment:
We have a `Helper` class. Why don't we move it there?

Reply:
moved